src/common.c

```diff
@@ -318,7 +320,20 @@ const char *ofi_straddr(char *buf, size_t *len,
 				str, *((uint16_t *)addr + 8), *((uint32_t *)addr + 5));
 		break;
 	case FI_SOCKADDR_IB:
-		size = snprintf(buf, *len, "fi_sockaddr_ib://%p", addr);
+		memset(str, 0, sizeof(str));
+		if (!inet_ntop(AF_INET6, ((uint64_t *)addr + 1), str, INET6_ADDRSTRLEN))
+			return NULL;
+
+		scope_id = ntohll(*((uint64_t *)addr + 5)) & 0xff;
+		sid = ntohll(*((uint64_t *)addr + 3));
+		size = snprintf(buf, *len, "fi_sockaddr_ib://[%s]" /* GID */
+			     ":%" PRIx16 /* P_Key */
+			     ":%" PRIx16 /* port space */
+			     ":%" PRIx8 /* Scope ID */,
```

Comment:
I looked at the existing code and I figured out that `FI_ADDR_IB_UD` was using hex and I decided to do the same. Let me know if you want me to change those fields.

Reply:
The pkey under /sys/class/blah blah is reported as hex.  So that makes sense.  I'm not sure about the other two.  As long as the output contains '0x' I guess it doesn't matter, since that makes it obvious.

Reply:
Do you mean that we should prefix all the hex fields with 0x?
Right now the output looks like this:
```fi_sockaddr_ib://[fe80::248a:703:1c:dc0d]:ffff:13f:0```

After converting the port space and scope id to be unsigned, the output would look like: ```fi_sockaddr_ib://[fe80::248a:703:1c:dc0d]:0xffff:319:0```?

Is that correct?

Reply:
Yes, I was suggesting that pkey (and other hex values) be written as 0xffff, rather than ffff.  That at least makes it clear that the value is hex.  This doesn't apply to the IP address itself, since the format is well defined.  We're trying to follow something close to the URI format.  I can't find any reference as to how URI handles hex versus decimal values, or if it's even defined.

So, assuming that there is NOT a URI standard for this, my vote is to report hex values with '0x' as the prefix.  This way, it doesn't matter if we use hex or decimal, we can support both, and the user can use whichever format is most convenient.

If this works, we can report port space and scope id either way.  I don't have a preference.  Sometime in the never going to happen future we can even convert those fields into a more readable string, like "tcp" or "udp" for port space, or "host" for scope.

Reply:
Thanks for your detailed explanation! I modified the patch so the user can specify both, hex values prefixed with '0x' and decimal values.